docs/dev/table/streaming/unbounded-data-processing/joins.md

```diff
@@ -22,17 +22,18 @@ specific language governing permissions and limitations
 under the License.
 -->
 
-Joins are a common and well-understood operation in batch data processing to connect the rows of two relations. However, the semantics of joins on [dynamic tables](dynamic_tables.html) are much less obvious or even confusing.
+Joins are a common and well-understood operation in batch data processing to connect the rows of two relations. However, the semantics of joins on [dynamic tables](dynamic_tables.html) representing unbounded data are fairly non-trivial. 
 
-Because of that, there are a couple of ways to actually perform a join using either Table API or SQL.
+However there are a couple of ways to actually perform a join using either Table API or SQL.
 
 For more information regarding the syntax, please check the join sections in [Table API](../tableApi.html#joins) and [SQL]({{ site.baseurl }}/dev/table/sql/queries.html#joins).
 
+//TODO: Explain the complications of unbounded joins
 * This will be replaced by the TOC
 {:toc}
 
-Regular Joins
--------------
+Regular Joins and its challenges
+---------------------------------------
 
```

Comment:
Lets get rid of "and its challenges". We want to make flink more approachable :)

Reply:
Sure.